File: fib/fib.js
#!/usr/bin/env node

const LEN = 10000

console.time('Array')

for (let _ = 0; _ < 100; _++) {
	let arr = new Uint32Array(LEN).map((_, i) => i * 2)

	console.log(`[0]=${arr[0]}`)
	console.log(`[1]=${arr[1]}`)
	console.log(`[${LEN - 1}]=${arr[LEN - 1]}`)
}

console.timeEnd('Array')

File: fib/fib.rs
#!/bin/sh
//usr/bin/env rustc -C opt-level=3 $0 -o a.out && ./a.out && rm ./a.out ; exit

use std::time::{Instant};

const LEN: usize = 10000;

fn main() {
    let now = Instant::now();

    for _ in 0..100 {
        let mut arr = [0; LEN];
        for i in arr {
            arr[i] = i * 2;
        }

        println!("[0]={}", arr[0]);
        println!("[1]={}", arr[1]);
        println!("[{}]={}", LEN - 1, arr[LEN - 1]);
    }
    
    println!("Array: {}ms", (now.elapsed().as_micros() as f64) / 1000.0);
}